Sources/GRPCNIOTransportHTTP2Posix/HTTP2ClientTransport+Posix.swift

```diff
@@ -105,7 +107,7 @@ extension HTTP2ClientTransport {
       self.channel.retryThrottle
     }
 
-    public func connect() async {
+    public func connect() async throws {
```

Comment:
Is it a regular Swift convention to keep the `throws` from the protocol signature if we don't actually throw in the implementation?

Reply:
I'm not sure if there is a convention. The `protocol` is `throws` so adding it back isn't an issue and leaves us scope in the future in case the implementation actually needs to become throwing.